### ISSUE TITLE
configure script fails

### ISSUE THREAD

**Issue by AntouanK** [OmegaPhil/hardware-monitor-applet#7]
I'm trying to install this panel, but it fails on ./configure

```
...
checking pkg-config is at least version 0.9.0... yes
checking for libxfce4panel-1.0 >= 4.8.0... not found
*** The required package libxfce4panel-1.0 was not found on your system.
*** Please install libxfce4panel-1.0 (atleast version 4.8.0) or adjust
*** the PKG_CONFIG_PATH environment variable if you
*** installed the package in a nonstandard prefix so that
*** pkg-config is able to find it.
```

I installed `libxfce4panel-2.0-dev` but that won't fix it.

Any ideas?

**Comment by OmegaPhil** [comment]
Please take a look at http://git.xfce.org/panel-plugins/xfce4-hardware-monitor-plugin/ (and the readme for the new issue tracker etc) - the repo here on github is from before the package was an official XFCE4 one. I'll see if I can close the issues now (and hopefully not lose access to them...).

Nope, issues disappear completely when they're turned off, so I guess this'll keep 'open' until all the Closed issues are completely obsolete.

**Comment by AntouanK** [comment]
I went to that page and got the 1.5.0.
But how is that going to help me solve that error I get on "configure"?
I tried googling the issue, but I cannot see any solution.

Is there any PPA that I can use?

**Comment by OmegaPhil** [comment]
OK, so you are using the latest code. I meant, this bug should be raised in the current bug tracker: https://bugzilla.xfce.org/enter_bug.cgi?product=Xfce4-hardware-monitor-plugin - I'll have a look at this tomorrow.

No PPA, but there are debs here: https://git.xfce.org/panel-plugins/xfce4-hardware-monitor-plugin/tree/1.5.0?h=omegaphil/pkg

**Comment by AntouanK** [comment]
Tried the .deb
Here's what I get

```
-> % sudo dpkg -i xfce4-hardware-monitor-plugin-dbgsym_1.5.0-1_amd64.deb
Selecting previously unselected package xfce4-hardware-monitor-plugin-dbgsym.
(Reading database ... 238412 files and directories currently installed.)
Preparing to unpack xfce4-hardware-monitor-plugin-dbgsym_1.5.0-1_amd64.deb ...
Unpacking xfce4-hardware-monitor-plugin-dbgsym (1.5.0-1) ...
dpkg: dependency problems prevent configuration of xfce4-hardware-monitor-plugin-dbgsym:
 xfce4-hardware-monitor-plugin-dbgsym depends on xfce4-hardware-monitor-plugin (= 1.5.0-1); however:
  Package xfce4-hardware-monitor-plugin is not installed.

dpkg: error processing package xfce4-hardware-monitor-plugin-dbgsym (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 xfce4-hardware-monitor-plugin-dbgsym
```

**Comment by OmegaPhil** [comment]
You need to install both the normal and debug packages at the same time, I usually use 'sudo dpkg -i *.deb' when the directory holds just the debs I want installed.

dbgsym is only useful if you want to debug segfaults and the like, you don't have to install it.

**Comment by AntouanK** [comment]
Ah yes, I didn't notice that.

Got the plain one now.
Tried again. 

```
-> % sudo dpkg -i xfce4-hardware-monitor-plugin_1.5.0-1_amd64.deb
Selecting previously unselected package xfce4-hardware-monitor-plugin.
(Reading database ... 238477 files and directories currently installed.)
Preparing to unpack xfce4-hardware-monitor-plugin_1.5.0-1_amd64.deb ...
Unpacking xfce4-hardware-monitor-plugin (1.5.0-1) ...
dpkg: dependency problems prevent configuration of xfce4-hardware-monitor-plugin:
 xfce4-hardware-monitor-plugin depends on libglibmm-2.4-1v5 (>= 2.48.0); however:
  Version of libglibmm-2.4-1v5:amd64 on system is 2.46.3-1.
 xfce4-hardware-monitor-plugin depends on libpangomm-1.4-1v5 (>= 2.40.0); however:
  Version of libpangomm-1.4-1v5:amd64 on system is 2.38.1-1.

dpkg: error processing package xfce4-hardware-monitor-plugin (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 xfce4-hardware-monitor-plugin
```

**Comment by OmegaPhil** [comment]
Hmm, the debs were built for Debian Testing a few months back., looks like what you have is too old. Whats the version of libxfce4panel (or similar name) on your system? You're using Debian Stable I guess?

**Comment by AntouanK** [comment]
I'm using Xubuntu.
`xfce-about` says I'm using 4.12
```
-> % sudo apt-get install libxfce4panel-2.0-4 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libxfce4panel-2.0-4 is already the newest version (4.12.0-3ubuntu2).
```

**Comment by OmegaPhil** [comment]
Do you have xfce4-panel-dev installed?

**Comment by AntouanK** [comment]
I just installed it.

Still I get the same errors on dependencies.

**Comment by OmegaPhil** [comment]
You need to give up on the debs, it looks like Ubuntu's versions are too old.

But your post demonstrates you haven't read the [README](https://git.xfce.org/panel-plugins/xfce4-hardware-monitor-plugin/tree/README), the required packages for building are under Requirements - make sure you've got those installed and try configuring again.

For reference I've just successfully configured on Devuan Testing, which is equivalent to Debian Testing.

**Comment by AntouanK** [comment]
No, I didn't read the whole README. I just checked I have xfce version > 4.4.
By the way, in the README it says nothing about `libglibmm` and `libpangomm`.
So all I have to do is find the latest of those two and install them manually?
I haven't installed libraries manually before, I always relied on the package manager.

**Comment by OmegaPhil** [comment]
apt-get/aptitude are package managers - there should be no need to install dependent debs manually. The requirements pull in the packages you mentioned by default (libgtkmm-2.4-dev depends on libglibmm and libglademm-2.4-1v5 depends on libpangomm).

**Comment by AntouanK** [comment]
but in xubuntu case it does pull earlier versions, right?
So I have to get the latest ones manually. Correct?

**Comment by OmegaPhil** [comment]
All I can say here is, from the perspective of building real debs, dependencies are worse/harder to satisfy - the control files produced have 'tight' dependencies - but building manually is much easier (I've found) since its satisfying dependencies based on configure.ac specification/pkg-config environment, e.g.:

```
dnl Configure the panel plugin (XFCE4 xdt-depends.m4)
XDT_CHECK_PACKAGE([LIBXFCE4PANEL], [libxfce4panel-1.0], [4.8.0])
XDT_CHECK_PACKAGE([LIBXFCE4UI], [libxfce4ui-1], [4.8.0])

dnl Checks for libraries
PKG_CHECK_MODULES([DEPS], [gtkmm-2.4 >= 2.24.0 \
                        libgnomecanvasmm-2.6 >= 2.6.0 \
                        libgtop-2.0 >= 2.6.0])
```

These required versions (from pkg-config) are very rarely bumped up since no genuinely new functionality should be used (this package's code is particularly ancient).

**Comment by OmegaPhil** [comment]
Basically, the dependencies required are ancient, so its unacceptable for Ubuntu to pull a package that is older than the requirement (when you are building yourself).

**Comment by AntouanK** [comment]
I found the latest versions in ubuntu launchpad
( e.g. https://launchpad.net/ubuntu/yakkety/amd64/libpangomm-1.4-1v5/2.40.1-3 )

But trying to install them is almost impossible. I keep getting errors and installing one uninstalls the other somehow.
I'll give up now. It's a shame cause this panel looks good.

Thanks for the help anyway.

**Comment by OmegaPhil** [comment]
OK, but you shouldn't be manually d/ling debs - aptitude (etc) will get suitable versions for you unless something is horribly broken.

**Comment by AntouanK** [comment]
But that's what the error messages say. 
It's unable to do it on it's own.
Am I missing something?

**Comment by AntouanK** [comment]
By the way, why does it need the latest versions? Is it really depending on something that exists only on the latest libs?

**Comment by OmegaPhil** [comment]
Building yourself does not rely on new packages. The prebuilt debs do since they're for Debian Testing.

**Comment by OmegaPhil** [comment]
What does this do for you:

```sudo aptitude install libglademm-2.4-dev libgnomecanvasmm-2.6-dev libgtkmm-2.4-dev libgtop2-dev libxfce4ui-1-dev xfce4-panel-dev```

**Comment by AntouanK** [comment]
From what I see I still get the same versions installed

```
Setting up libglade2-0:amd64 (1:2.6.4-2) ...
Setting up libgail18:amd64 (2.24.30-1ubuntu1) ...
Setting up libgnomecanvas2-common (2.30.3-2) ...
Setting up libgnomecanvas2-0:amd64 (2.30.3-2) ...
Setting up libgtop2-common (2.32.0-1) ...
Setting up libgtop-2.0-10:amd64 (2.32.0-1) ...
Setting up gir1.2-gtop-2.0:amd64 (2.32.0-1) ...
Setting up icu-devtools (55.1-7) ...
Setting up libart-2.0-dev (2.3.21-2) ...
Setting up libglibmm-2.4-1v5:amd64 (2.46.3-1) ...
Setting up libatkmm-1.6-1v5:amd64 (2.24.2-1) ...
Setting up libsigc++-2.0-dev:amd64 (2.6.2-1) ...
Setting up libglibmm-2.4-dev:amd64 (2.46.3-1) ...
Setting up libatkmm-1.6-dev:amd64 (2.24.2-1) ...
Setting up libcairomm-1.0-dev:amd64 (1.12.0-1) ...
Setting up libgail-common:amd64 (2.24.30-1ubuntu1) ...
Setting up libgail-dev (2.24.30-1ubuntu1) ...
Setting up libicu-dev:amd64 (55.1-7) ...
Setting up libxml2-dev:amd64 (2.9.3+dfsg1-1ubuntu0.1) ...
Setting up libglade2-dev:amd64 (1:2.6.4-2) ...
Setting up libpangomm-1.4-1v5:amd64 (2.38.1-1) ...
Setting up libgtkmm-2.4-1v5:amd64 (1:2.24.4-2) ...
Setting up libglademm-2.4-1v5 (2.6.7-5) ...
Setting up libpangomm-1.4-dev:amd64 (2.38.1-1) ...
Setting up libgtkmm-2.4-dev:amd64 (1:2.24.4-2) ...
Setting up libglademm-2.4-dev (2.6.7-5) ...
Setting up libgnomecanvas2-dev:amd64 (2.30.3-2) ...
Setting up libgnomecanvasmm-2.6-1v5 (2.26.0-2) ...
Setting up libgnomecanvasmm-2.6-dev (2.26.0-2) ...
Setting up libgtop2-dev:amd64 (2.32.0-1) ...
Setting up libxfce4ui-1-dev (4.12.1-2) ...
Processing triggers for libc-bin (2.23-0ubuntu5) ...
```

**Comment by OmegaPhil** [comment]
So that looks correct, can I have the full ./configure output now?

**Comment by AntouanK** [comment]
Worked fine now

```
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking for library containing strerror... none required
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking whether NLS is requested... yes
checking for intltool >= 0.40.0... 0.51.0 found
checking for intltool-update... /usr/bin/intltool-update
checking for intltool-merge... /usr/bin/intltool-merge
checking for intltool-extract... /usr/bin/intltool-extract
checking for xgettext... /usr/bin/xgettext
checking for msgmerge... /usr/bin/msgmerge
checking for msgfmt... /usr/bin/msgfmt
checking for gmsgfmt... /usr/bin/msgfmt
checking for perl... /usr/bin/perl
checking for perl >= 5.8.1... 5.22.1
checking for XML::Parser... ok
checking for ar... ar
checking the archiver (ar) interface... ar
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /bin/sed
checking for fgrep... /bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /bin/dd
checking how to truncate binary pipes... /bin/dd bs=4096 count=1
checking for mt... mt
checking if mt is a manifest tool... no
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... yes
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking for ANSI C header files... (cached) yes
checking pkg-config is at least version 0.9.0... yes
checking for libxfce4panel-1.0 >= 4.8.0... 4.12.0
checking LIBXFCE4PANEL_CFLAGS... -pthread -I/usr/include/xfce4/libxfce4panel-1.0 -I/usr/include/gtk-2.0 -I/usr/lib/x86_64-linux-gnu/gtk-2.0/include -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/libpng12 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng12 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/freetype2 -I/usr/include/xfce4 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include
checking LIBXFCE4PANEL_LIBS... -lxfce4panel-1.0 -lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lfontconfig -lfreetype -Wl,--export-dynamic -lgmodule-2.0 -pthread -lxfce4util -lglib-2.0
checking pkg-config is at least version 0.9.0... yes
checking for libxfce4ui-1 >= 4.8.0... 4.12.1
checking LIBXFCE4UI_CFLAGS... -pthread -I/usr/include/xfce4/libxfce4ui-1 -I/usr/include/gtk-2.0 -I/usr/lib/x86_64-linux-gnu/gtk-2.0/include -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/libpng12 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng12 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/freetype2 -I/usr/include/xfce4 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include
checking LIBXFCE4UI_LIBS... -lxfce4ui-1 -lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lfontconfig -lfreetype -lxfce4util -lglib-2.0
checking for DEPS... yes
checking locale.h usability... yes
checking locale.h presence... yes
checking for locale.h... yes
checking for LC_MESSAGES... yes
checking libintl.h usability... yes
checking libintl.h presence... yes
checking for libintl.h... yes
checking for ngettext in libc... yes
checking for dgettext in libc... yes
checking for bind_textdomain_codeset... yes
checking for msgfmt... (cached) /usr/bin/msgfmt
checking for dcgettext... yes
checking if msgfmt accepts -c... yes
checking for gmsgfmt... (cached) /usr/bin/msgfmt
checking for xgettext... (cached) /usr/bin/xgettext
checking for catalogs to be installed...  ar as az be bg bn_IN bs ca cs cy da de dz el en_CA en_GB es eu fi fr gl gu he hi hr hu it ja kn ko lv mr ms nb ne nl oc pa pl pt pt_BR ru rw sl sq sr sr@Latn sv ta th tr uk vi zh_CN zh_HK zh_TW
checking for bind_textdomain_codeset... (cached) yes
checking for locales directory... ${datarootdir}/locale
checking for additional xgettext flags... --keyword=Q_ --from-code=UTF-8
checking sensors/sensors.h usability... no
checking sensors/sensors.h presence... no
checking for sensors/sensors.h... no
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating po/Makefile.in
config.status: creating config.h
config.status: config.h is unchanged
config.status: executing depfiles commands
config.status: executing libtool commands
config.status: executing default-1 commands
config.status: executing po/stamp-it commands
```

**Comment by OmegaPhil** [comment]
So yes, thats what the README was telling you to do (once you realised that the provided debs weren't compatible with your distro).

**Comment by AntouanK** [comment]
So using `aptitude` instead of `apt-get` was the issue?!?

**Comment by OmegaPhil** [comment]
I know their dependency stuff is a little different, but it shouldn't have made a difference here?

**Comment by AntouanK** [comment]
I don't even know what the difference is on those two.
I'll have a look.

In the meantime I did `make` and `make install`.
I get a `xfce4-hardware-monitor-plugin.desktop` file created.
And I copy that into `/usr/share/xfce4/panel/plugins/` where the rest of the plugins are.
It should now come up in the "add panel" list, right?

**Comment by OmegaPhil** [comment]
From the README its './configure --prefix=/usr', and 'sudo make install' should put all files in a usable place (so the panel should notice the new .desktop file and then list the plugin).

If it starts dumping under /usr/local, the prefix is wrong and the panel won't notice the new plugin.

**Comment by AntouanK** [comment]
I did `'./configure --prefix=/usr` and `make install`, it didn't put it in the folder I mentioned.

But even after I copied it manually, I cannot see it in the list of panels.
Do I have to restart xfce?

**Comment by OmegaPhil** [comment]
What is the full output of make install?

**Comment by AntouanK** [comment]
My bad, `sudo make install` does copy the file correctly. I can now see it in the list.
It works fine. 
Thanks for the help!  :+1:

**Comment by AntouanK** [comment]
A couple of questions ( last ones, I promise ),
- can we display the value somewhere next to the bars? or at least the "tag"?
- how do we set the max value for the graph to scale to?

**Comment by OmegaPhil** [comment]
Please open up an issue on the real bug tracker for these - remember this Github one is not technically open.

**Comment by AntouanK** [comment]
Ok. The link for that bug tracker?

The things I asked, they are not bugs, right? Just some extra features. But maybe they are already there, that's why I asked. I cannot see any features listed in the README. Maybe I missed them.

**Comment by OmegaPhil** [comment]
I've already given you the link before, when I asked you to move this bug there: https://bugzilla.xfce.org/enter_bug.cgi?product=Xfce4-hardware-monitor-plugin

Yes, but its useful to others that might have the question. This is the real documentation page: http://goodies.xfce.org/projects/panel-plugins/xfce4-hardware-monitor-plugin
